manager/controllers/app/metadata/data_info.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+package metadata
```

Comment:
Not a great name.  Metadata about what?  workload, infra, data?  Perhaps datametadata?  
Why isn't all of this part of the catalog connector?  I know in this PR you are just moving code around, so perhaps open an issue about this.